packages/core/src/configs/node/jest.setup.js

```diff
@@ -1,23 +1,108 @@
 const path = require('path')
 
-const camelCase = require('lodash/camelCase')
-
 const { getPaths } = require('@redwoodjs/internal')
+const { defineScenario } = require('@redwoodjs/testing/dist/scenario')
+const { db } = require(path.join(getPaths().api.src, 'lib', 'db'))
+const DEFAULT_SCENARIO = 'standard'
+const PRISMA_RESERVED = ['create', 'connect']
 
-const redwoodPaths = getPaths()
+const findNestedModels = (data) => {
+  let models = []
 
-const { db } = require(path.join(redwoodPaths.api.src, 'lib', 'db'))
+  for (const [field, value] of Object.entries(data)) {
+    if (typeof value === 'object') {
+      if (!models.includes(field) && !PRISMA_RESERVED.includes(field)) {
+        models.push(field)
+      }
+      models = models.concat(findNestedModels(value))
+    }
+  }
 
-/**
- * Loop through all db models and clear the tables before every test
- */
-beforeEach(async () => {
-  for (let model of db._dmmf.datamodel.models) {
-    if (model.name === 'RW_DataMigration') continue
+  return models
+}
 
-    await db[camelCase(model.name)].deleteMany()
+const seedScenario = async (scenario) => {
+  if (scenario) {
+    const scenarios = {}
+    for (const [model, namedFixtures] of Object.entries(scenario)) {
+      scenarios[model] = {}
+      for (const [name, data] of Object.entries(namedFixtures)) {
+        scenarios[model][name] = await db[model].create({ data })
+      }
+    }
+    return scenarios
+  } else {
+    return {}
   }
-})
+}
+
+const removeScenario = async (scenario) => {
+  if (scenario) {
+    let models = []
+
+    for (const [model, namedFixtures] of Object.entries(scenario)) {
+      models.push(model)
+      for (const [_name, data] of Object.entries(namedFixtures)) {
+        models = models.concat(findNestedModels(data))
+      }
+    }
+    // get unique model names only
+    models = Array.from(new Set(models))
+
+    for (const model of models) {
+      await db.$queryRaw(`DELETE FROM ${model}`)
+    }
+  }
+}
+
+window.scenario = (...args) => {
+  let scenarioName, testName, testFunc
+
+  if (args.length === 3) {
+    ;[scenarioName, testName, testFunc] = args
+  } else {
+    scenarioName = DEFAULT_SCENARIO
+    ;[testName, testFunc] = args
+  }
```

Comment:
Maybe we should check if it's two args?

and if it isn't then we should throw an error?

Reply:
Check out this change, I explicitly check for 2 and if not then we throw an error letting people know it needs 2 or 3 only.